src/util/MemorySize/MemorySize.h

```diff
@@ -152,25 +152,34 @@ amount of bytes per unit.
 */
 template <typename T>
 requires std::integral<T> || std::floating_point<T>
-size_t constexpr convertMemoryUnitsToBytes(const T amountOfUnits,
+constexpr size_t convertMemoryUnitsToBytes(const T amountOfUnits,
                                            const size_t numBytesPerUnit) {
   // Negativ values makes no sense.
   AD_CONTRACT_CHECK(amountOfUnits >= 0);
 
   // Max value for `amountOfUnits`.
-  AD_CONTRACT_CHECK(static_cast<T>(sizeTDivision(
-                        size_t_max, numBytesPerUnit)) >= amountOfUnits);
+  if (static_cast<T>(sizeTDivision(size_t_max, numBytesPerUnit)) <
+      amountOfUnits) {
+    throw std::runtime_error(
+        "The given amount of memory is more memory, than can be represented "
+        "and worked with in c++.");
+  }
```

Comment:
Please somehow integrate the number in the error message so like
`429123049146801876912358 TB cannot be stored in 64 bits when being converted to bytes`

Reply:
(I Think it's easiest to also pass the units `string_view` to this function.

Reply:
What `string_view`? All the parameter are numbers.

Reply:
Done, see [bdc60a4](https://github.com/ad-freiburg/qlever/pull/1032/commits/bdc60a4dd795827cd878dd4b303e232b261d244b).